### ISSUE TITLE
Missing Volume Icon

### ISSUE THREAD

**Issue by MggMuggins** [pithos/pithos#525]
I'm using latest Git from the AUR. The volume icon is a question mark. I'm on Manjaro KDE. Is this just an issue trying to access the wrong resource via GTK, or something on my end?

**Comment by TingPing** [comment]
I've seen this periodically happen in a few of my applications, but restarting the application usually solves it.

**Comment by JasonLG1979** [comment]
I've noticed the same thing. Every now and then I get a the "missing icon" icon. It seems pretty random and the theme doesn't seem to matter either. I just chalked it up to a GTK bug. I want to say it started for me with 3.24?

**Comment by MggMuggins** [comment]
I can confirm that restarting the app fixes the icon. I agree, it's probably upstream...

**Comment by JasonLG1979** [comment]
@TingPing setting the use_symbolic prop on the volume button to False seems to solve the problem so far in my tests. My theory is that since we're setting the icon names to *-symbolic already maybe Gtk is trying to find *-symbolic-symbolic and ofc failing that's why we're getting the broken image icon.

**Comment by TingPing** [comment]
To the source! https://github.com/GNOME/gtk/blob/gtk-3-22/gtk/gtkvolumebutton.c

It seems to work like this:

- If `icons` property is set those are used.
- If `use-symbolic` property is set, it sets `icons` property for you.
- If you query `use-symbolic` it actually queries does `icons` end in `-symbolic`.

So what we should be doing I believe is not setting `icons` and simply set `use-symbolic` to `True`.

**Comment by JasonLG1979** [comment]
> If use-symbolic property is set, it sets icons property for you

use-symbolic defaults to True according to the PyGObject docs.

**Comment by TingPing** [comment]
Indeed. So on construction we set `icons` to symbolic icons and it sets `use-symbolic` to `true`.

Worth noting the default value is of `icons` is indeed `image-missing`: https://github.com/GNOME/gtk/blob/gtk-3-22/gtk/gtkscalebutton.c#L988-L994

So order in which these are set matters but afaict both orders should *work* fine.

**Comment by JasonLG1979** [comment]
In any event setting that prop is unnecessary. I put in a PR to remove it.

**Comment by JasonLG1979** [comment]
> So order in which these are set matters but afaict both orders should work fine.

Weird race condition?

**Comment by TingPing** [comment]
> Weird race condition?

Well its single threaded and as far as I can tell both orders are valid:

- If we set `icons` first it should update the image and `use-symbolic=true` does nothing
- If we set `use-symbolic` first it should update the images and then setting `icons` next just updates the image again.

**Comment by JasonLG1979** [comment]
Something's going wonky? In any case there's no need to set the prop if it's preset, it's just redundant.

**Comment by TingPing** [comment]
Probably need to run it under a debugger to see what some values are and figure out whats not triggering.
